gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,593 @@
+    project.setCreateTime(ISODateTimeFormat.dateTime().print(System.currentTimeMillis()));
+    Response response;
+    String customErrorMessage = checkForProjectErrors(project);
+    if (customErrorMessage != null) {
+      response = Error.INVALID_ARGUMENT.response(customErrorMessage);
+    } else if (projects.containsKey(project.getProjectId())) {
+      response = Error.ALREADY_EXISTS.response(
+          "A project with the same project ID (" + project.getProjectId() + ") already exists.");
+    } else {
+      projects.put(project.getProjectId(), project);
+      try {
+        String createdProjectStr = jsonFactory.toString(project);
+        response = new Response(HttpURLConnection.HTTP_OK, createdProjectStr);
+      } catch (IOException e) {
+        response =
+            Error.INTERNAL_ERROR.response("Error serializing project " + project.getProjectId());
+      }
+    }
+    return response;
+  }
+
+  Response delete(String projectId) {
+    Project project = projects.get(checkNotNull(projectId));
+    Response response;
+    if (project == null) {
+      // when possible, change this to 404 (#440)
+      response = Error.PERMISSION_DENIED.response(
+          "Error when deleting " + projectId + " because the project was not found.");
+    } else if (!project.getLifecycleState().equals("ACTIVE")) {
+      response = Error.FAILED_PRECONDITION.response(
+          "Error when deleting " + projectId + " because the lifecycle state was not ACTIVE.");
+    } else {
+      project.setLifecycleState("DELETE_REQUESTED");
+      response = new Response(HttpURLConnection.HTTP_OK, "{}");
+    }
+    return response;
+  }
+
+  Response get(String projectId, String[] fields) {
+    if (!projects.containsKey(checkNotNull(projectId))) {
+      // when possible, change this to 404 (#440)
+      return Error.PERMISSION_DENIED.response("Project " + projectId + " not found.");
+    }
+    Project project = projects.get(projectId);
+    try {
+      return new Response(
+          HttpURLConnection.HTTP_OK, jsonFactory.toString(extractFields(project, fields)));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(
+          "Error when serializing project " + project.getProjectId());
+    }
+  }
+
+  Response list(final Map<String, Object> options) {
+    // Use pageSize and pageToken options when Cloud Resource Manager does so (#421)
+    List<String> projectsSerialized = new ArrayList<>();
+    for (Project p : projects.values()) {
+      Boolean includeProject = includeProject(p, (String[]) options.get("filter"));
+      if (includeProject) {
+        try {
+          projectsSerialized.add(
+              jsonFactory.toString(extractFields(p, (String[]) options.get("fields"))));
+        } catch (IOException e) {
+          return Error.INTERNAL_ERROR.response(
+              "Error when serializing project " + p.getProjectId());
+        }
+      } else if (includeProject == null) {
```

Comment:
Actually I see now what you are doing. You were using the `null` as a third value to indicate bad request. I think that is confusing and should be avoided. Plus, I am sure there is no test for it as I would expect the unboxing in line 321 "if (includeProject)" to trigger NPE in that case.

As suggested above get the filter before the loop, run a check on it and then change includeProject to return `boolean`.